opensrp-anc/src/main/java/org/smartregister/anc/library/repository/PatientRepository.java

```diff
@@ -108,20 +108,23 @@ public static void updateWomanAlertStatus(String baseEntityId, String alertStatu
         ContentValues contentValues = new ContentValues();
         contentValues.put(DBConstantsUtils.KeyUtils.CONTACT_STATUS, alertStatus);
 
-        getMasterRepository().getWritableDatabase()
-                .update(getRegisterQueryProvider().getDetailsTable(), contentValues, DBConstantsUtils.KeyUtils.BASE_ENTITY_ID + " = ?",
-                        new String[]{baseEntityId});
+        updatePatient(baseEntityId, contentValues, getRegisterQueryProvider().getDetailsTable());
 
         updateLastInteractedWith(baseEntityId);
     }
 
+    public static void updatePatient(String baseEntityId, ContentValues contentValues, String detailsTable) {
```

Comment:
If `detailsTable` is always the same, do the calling functions really need to always specify it? 

i.e. could the previous `getRegisterQueryProvider().getDetailsTable()` still work?

Reply:
Its doesn't do the updates for the details table alone. `org.smartregister.anc.library.repository.PatientRepository#updateLastInteractedWith` here we are setting the demographics table. I have updated the name to be more generic.